cli/flox-rust-sdk/src/providers/build.rs

```diff
@@ -161,3 +161,435 @@ fn read_output_to_channel(
+
+    #[test]
+    fn build_returns_failure_when_package_not_defined() {
+        let package_name = String::from("foo");
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, "version = 1");
+
+        assert_build_status(&flox, &mut env, &package_name, false);
+    }
+
+    #[test]
+    fn build_command_generates_file() {
+        let package_name = String::from("foo");
+        let file_name = String::from("bar");
+        let file_content = String::from("some content");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            command = """
+                mkdir $out
+                echo -n {file_content} > $out/{file_name}
+            """
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+        let env_path = env.parent_path().unwrap();
+
+        assert_build_status(&flox, &mut env, &package_name, true);
+        assert_build_file(&env_path, &package_name, &file_name, &file_content);
+    }
+
+    #[test]
+    #[ignore] // TODO: `files` isn't currently passed to or parsed by `flox-build.mk`.
+    fn build_includes_files() {
+        let package_name = String::from("foo");
+        let file_name = String::from("bar");
+        let file_content = String::from("some content");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            files = ["{file_name}"]
+            command = "mkdir $out"
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+        let env_path = env.parent_path().unwrap();
+
+        fs::write(env_path.join(&file_name), &file_content).unwrap();
+
+        assert_build_status(&flox, &mut env, &package_name, true);
+        assert_build_file(&env_path, &package_name, &file_name, &file_content);
+    }
+
+    #[test]
+    #[ignore] // TODO: `systems` isn't currently passed to or parsed by `flox-build.mk`.
+    fn build_restricts_systems() {
+        let package_name = String::from("foo");
+        let file_name = String::from("bar");
+        let file_content = String::from("some content");
+
+        let manifest = formatdoc! {r#"
+            version = 1
+
+            [build.{package_name}]
+            systems = ["invalid"]
+            command = """
+                mkdir $out
+                echo -n {file_content} > $out/{file_name}
+            """
+        "#};
+
+        let (flox, _temp_dir_handle) = flox_instance();
+        let mut env = new_path_environment(&flox, &manifest);
+        let env_path = env.parent_path().unwrap();
+
+        fs::write(env_path.join(&file_name), &file_content).unwrap();
+
+        assert_build_status(&flox, &mut env, &package_name, false);
+        let dir = result_dir(&env_path, &package_name);
+        assert!(!dir.exists());
+    }
```

Comment:
I'm unsure whether the `systems` attribute is _supposed_ to be checked by the builder at all.
So far i've seen this more as a cli side check like systems on the `install`, and `services` tables.

Reply:
I think I'd agree the systems attribute validation should all happen well before the builder is invoked.

Reply:
This is tricky when taking build composition into account.
We would need to know which other builds a build depends on _in rust_ and build a similar graph as the makefile does to ensure a build does not advertise a broader set of valid systems than its dependencies. Not sure if we're cool with that degree of introspection. L